### ISSUE TITLE
Include IP address and originating script

### ISSUE THREAD

**Issue by bsmoo** [No3x/wp-mail-logging#42]
To help detect spam, please can you include the IP address that sent the mail request and the originating script.

**Comment by No3x** [comment]
> please can you include the IP address that sent the mail

It's always the IP of your hosts WP is running on.

> the originating script.

This information is not provided by WP

**Comment by bsmoo** [comment]
If Wordpress is running over 6 servers, it would be great to see which one actually fired off the email. Will help with troubleshooting non-delivery.

Script no problem, this is usually injected into the mail headers anyway with `X-php-originating-script`

**Comment by No3x** [comment]
Well sure. I did not expect such a "advanced" setup.
So $_SERVER['SERVER_ADDR'] would be sufficient?

**Comment by bsmoo** [comment]
That would definitely suffice.

**Comment by No3x** [comment]
Is it actually some kind of load balancing? So the WordPress instances are the same on each server and they share a database? 
(looking for a mechanism to avoid storing the IP for each mail)

**Comment by bsmoo** [comment]
It is a load balanced solution, sharing the same database.

**Comment by No3x** [comment]
Unfortunately there is no way to avoid this overhead for each mail..
But it's really simple to implement.

**Comment by No3x** [comment]
I'm currently working on this.
Do you think it is better to store the hostname instead of the ip?

**Comment by No3x** [comment]
I have created a beta. Let me know if you want to test it:
https://github.com/No3x/wp-mail-logging/releases/tag/beta%2F1.7.0R1

**Comment by bsmoo** [comment]
Great, I'll test this today/tomorrow

**Comment by No3x** [comment]
@bsmoo was the test successful?

**Comment by bsmoo** [comment]
Sorry about that, forgot to get to this one... Where is it supposed to be displayed?

```
Time:
2016-10-23 12:54:40
Receiver:
spam@example.co.uk
Subject:
fsdgf
Message:
Message Body: gfdgdfg -- This e-mail was sent from a contact form on example (http://example.co.uk)
Headers:
From: example <spam@example.co.uk> X-WPCF7-Content-Type: text/plain
Attachments:
```

**Comment by No3x** [comment]
You need to enable it in the settings first.

**Comment by bsmoo** [comment]
Looks great to me!
Thanks!

**Comment by No3x** [comment]
Nice! 
I will push this to the WordPress repo soon. You should not have problems switching to the version from the WordPress repo if I don't change the database schema until release of 1.7.0.

**Comment by No3x** [comment]
I have just released 1.7.0. The update should bot be detected by WordPress if you still use that beta version. On the next release there will be an update as usual.
If you want to remove the beta and use the one from WordPress.org just remove it and install via WordPress.
